# PAI


<!-- WARNING: THIS FILE WAS AUTOGENERATED! DO NOT EDIT! -->

------------------------------------------------------------------------

<a
href="https://github.com/mtinti/protein_cutter/blob/main/protein_cutter/annotation.py#L61"
target="_blank" style="float:right; font-size:smaller">source</a>

### get_observable

>  get_observable (sequence:str, enzyme:str='trypsin_full')

*Get the set of observable peptides from in silico digestion.*

Performs in silico digestion of a protein sequence and returns the set
of theoretically observable peptides. This is used as the denominator
(N_observable) in PAI and emPAI calculations.

<table>
<colgroup>
<col style="width: 6%" />
<col style="width: 25%" />
<col style="width: 34%" />
<col style="width: 34%" />
</colgroup>
<thead>
<tr>
<th></th>
<th><strong>Type</strong></th>
<th><strong>Default</strong></th>
<th><strong>Details</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>sequence</td>
<td>str</td>
<td></td>
<td>Protein amino acid sequence to digest.</td>
</tr>
<tr>
<td>enzyme</td>
<td>str</td>
<td>trypsin_full</td>
<td>Enzyme name for in silico digestion. Must be a key
in<br>protease_dict. Default is trypsin_full.</td>
</tr>
<tr>
<td><strong>Returns</strong></td>
<td><strong>set</strong></td>
<td></td>
<td><strong>Set of unique observable peptide sequences after
digestion<br>and filtering by m/z range.</strong></td>
</tr>
</tbody>
</table>

------------------------------------------------------------------------

<a
href="https://github.com/mtinti/protein_cutter/blob/main/protein_cutter/pai.py#L118"
target="_blank" style="float:right; font-size:smaller">source</a>

### add_pai

>  add_pai (prot_df_path:Union[str,pathlib.Path],
>               pep_df_path:Union[str,pathlib.Path],
>               fasta_path:Union[str,pathlib.Path],
>               output_path:Union[str,pathlib.Path]=None,
>               pg_in_prot:str='PG.ProteinGroups',
>               pg_in_pep:str='PG.ProteinAccessions',
>               missed_cleavages_col:str='PEP.NrOfMissedCleavages',
>               pep_stripped_col:str='PEP.StrippedSequence',
>               enzyme:str='trypsin_full', filter_missed_cleavages:bool=True)

*Calculate Protein Abundance Index (PAI) for each protein group.*

Computes the ratio of observed peptides to observable peptides for each
protein, providing a measure of sequence coverage that normalizes for
protein size. This metric is useful for distinguishing between
peptide-level and protein-level evidence in proteomics experiments.

<table>
<colgroup>
<col style="width: 6%" />
<col style="width: 25%" />
<col style="width: 34%" />
<col style="width: 34%" />
</colgroup>
<thead>
<tr>
<th></th>
<th><strong>Type</strong></th>
<th><strong>Default</strong></th>
<th><strong>Details</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>prot_df_path</td>
<td>Union</td>
<td></td>
<td>Path to protein-level report file (TSV format).</td>
</tr>
<tr>
<td>pep_df_path</td>
<td>Union</td>
<td></td>
<td>Path to peptide-level report file (TSV format).</td>
</tr>
<tr>
<td>fasta_path</td>
<td>Union</td>
<td></td>
<td>Path to FASTA file containing protein sequences.</td>
</tr>
<tr>
<td>output_path</td>
<td>Union</td>
<td>None</td>
<td>Path for output file. If None, appends ‘.pai’ before
the<br>extension of prot_df_path. Default is None.</td>
</tr>
<tr>
<td>pg_in_prot</td>
<td>str</td>
<td>PG.ProteinGroups</td>
<td>Column name for protein groups in protein report.<br>Default is
PG.ProteinGroups.</td>
</tr>
<tr>
<td>pg_in_pep</td>
<td>str</td>
<td>PG.ProteinAccessions</td>
<td>Column name for protein accessions in peptide report.<br>Default is
PG.ProteinAccessions.</td>
</tr>
<tr>
<td>missed_cleavages_col</td>
<td>str</td>
<td>PEP.NrOfMissedCleavages</td>
<td>Column name for missed cleavages count in peptide report.<br>Default
is PEP.NrOfMissedCleavages.</td>
</tr>
<tr>
<td>pep_stripped_col</td>
<td>str</td>
<td>PEP.StrippedSequence</td>
<td>Column name for stripped peptide sequences.<br>Default is
PEP.StrippedSequence.</td>
</tr>
<tr>
<td>enzyme</td>
<td>str</td>
<td>trypsin_full</td>
<td>Enzyme name for in silico digestion to calculate
observable<br>peptides. Default is trypsin_full.</td>
</tr>
<tr>
<td>filter_missed_cleavages</td>
<td>bool</td>
<td>True</td>
<td>If True, only consider peptides with zero missed cleavages<br>for
observed count. Default is True.</td>
</tr>
<tr>
<td><strong>Returns</strong></td>
<td><strong>DataFrame</strong></td>
<td></td>
<td><strong>Protein report DataFrame with added pai column
containing<br>the ratio of observed to observable
peptides.</strong></td>
</tr>
</tbody>
</table>

``` python
add_pai(prot_df_path = TEST_DATA/'pipeline_test/prot_report.tsv',
    pep_df_path =  TEST_DATA/'pipeline_test/pep_report.tsv',
    fasta_path = TEST_DATA/'pipeline_test/prot.fa')
```

    /Users/mtinti/git_projects/protein_cutter/nbs

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
&#10;    .dataframe tbody tr th {
        vertical-align: top;
    }
&#10;    .dataframe thead th {
        text-align: right;
    }
</style>

<table class="dataframe" data-quarto-postprocess="true" data-border="1">
<thead>
<tr style="text-align: right;">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">PG.ProteinGroups</th>
<th data-quarto-table-cell-role="th">[1]
331_2022_GBR-Cell-34C-A.raw.PG.Quantity</th>
<th data-quarto-table-cell-role="th">[2]
331_2022_GBR-Cell-34C-B.raw.PG.Quantity</th>
<th data-quarto-table-cell-role="th">[3]
331_2022_GBR-Cell-34C-C.raw.PG.Quantity</th>
<th data-quarto-table-cell-role="th">[4]
331_2022_GBR-Cell-37C-A.raw.PG.Quantity</th>
<th data-quarto-table-cell-role="th">[5]
331_2022_GBR-Cell-37C-B.raw.PG.Quantity</th>
<th data-quarto-table-cell-role="th">[6]
331_2022_GBR-Cell-37C-C.raw.PG.Quantity</th>
<th data-quarto-table-cell-role="th">[7]
331_2022_GBR-Cell-40C-A.raw.PG.Quantity</th>
<th data-quarto-table-cell-role="th">[8]
331_2022_GBR-Cell-40C-B.raw.PG.Quantity</th>
<th data-quarto-table-cell-role="th">...</th>
<th data-quarto-table-cell-role="th">[10]
331_2022_GBR-SN-34C-A.raw.PG.Quantity</th>
<th data-quarto-table-cell-role="th">[11]
331_2022_GBR-SN-34C-B.raw.PG.Quantity</th>
<th data-quarto-table-cell-role="th">[12]
331_2022_GBR-SN-34C-C.raw.PG.Quantity</th>
<th data-quarto-table-cell-role="th">[13]
331_2022_GBR-SN-37C-A.raw.PG.Quantity</th>
<th data-quarto-table-cell-role="th">[14]
331_2022_GBR-SN-37C-B.raw.PG.Quantity</th>
<th data-quarto-table-cell-role="th">[15]
331_2022_GBR-SN-37C-C.raw.PG.Quantity</th>
<th data-quarto-table-cell-role="th">[16]
331_2022_GBR-SN-40C-A.raw.PG.Quantity</th>
<th data-quarto-table-cell-role="th">[17]
331_2022_GBR-SN-40C-B.raw.PG.Quantity</th>
<th data-quarto-table-cell-role="th">[18]
331_2022_GBR-SN-40C-C.raw.PG.Quantity</th>
<th data-quarto-table-cell-role="th">pai</th>
</tr>
</thead>
<tbody>
<tr>
<td data-quarto-table-cell-role="th">0</td>
<td>0</td>
<td>Tb427_000008800.1-p1</td>
<td>551307.562500</td>
<td>704164.562500</td>
<td>748118.875000</td>
<td>681518.625000</td>
<td>703385.687500</td>
<td>686961.1250</td>
<td>816417.812500</td>
<td>631893.750000</td>
<td>...</td>
<td>2.036847e+05</td>
<td>189815.296875</td>
<td>83305.46875</td>
<td>2.046948e+05</td>
<td>167546.3125</td>
<td>117179.421875</td>
<td>215901.4375</td>
<td>273887.625</td>
<td>135870.109375</td>
<td>0.153846</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">1</td>
<td>29</td>
<td>Tb427_000198100.1-p1</td>
<td>17201.074219</td>
<td>22934.291016</td>
<td>24707.408203</td>
<td>20580.830078</td>
<td>23038.123047</td>
<td>24520.1875</td>
<td>29426.867188</td>
<td>28961.535156</td>
<td>...</td>
<td>1.619666e+06</td>
<td>787221.187500</td>
<td>106877.75000</td>
<td>1.162788e+06</td>
<td>311610.7500</td>
<td>151712.437500</td>
<td>565188.1875</td>
<td>584564.250</td>
<td>379278.843750</td>
<td>0.318841</td>
</tr>
</tbody>
</table>

<p>2 rows × 21 columns</p>
</div>

``` python
# pipline
# flag_proprietary_from_pg protein to filter out uniprots
# flag_proprietary_from_pg peptide to filter out uniprots
# load filtter prot and pep dataset
# load fasta 
# add pai to protein group
```
